rust/chains/hyperlane-cosmos/src/providers/mod.rs

```diff
@@ -89,7 +90,7 @@ impl HyperlaneProvider for CosmosProvider {
     }
 
     async fn is_contract(&self, _address: &H256) -> ChainResult<bool> {
-        // FIXME
+        let _ = self.grpc_client.wasm_contract_info().await?;
```

Comment:
should this value be returned here?

Reply:
I took another look. It is better to report `Ok(false)` on error here since otherwise the message will be re-tried on prepare. I added a commit.